### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by Pialex99**

### CODE REVIEW

File: app/src/main/java/ch/epfl/balelecbud/model/EmergencyInformation.java [hunk 1]
```diff
@@ -2,15 +2,30 @@
...
-
-    }
-
+    /**
+     * Empty constructor used by FireStore
+     */
+    public EmergencyInformation() { }
+
+    /**
+     * Constructor for emergencies information
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/model/EmergencyInformation.java:22]: emergency information

File: app/src/main/java/ch/epfl/balelecbud/model/MarkerType.java [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 import ch.epfl.balelecbud.R;
 
+/**
+ * Enum modeling the different marker type
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/model/MarkerType.java:6]: types*

File: app/src/main/java/ch/epfl/balelecbud/model/MyMarker.java [hunk 3]
```diff
@@ -12,43 +12,94 @@
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppContext;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * Interface modeling marker on the map
+ */
 public interface MyMarker {
 
+    /**
+     * Change the marker current location
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/model/MyMarker.java:21]: marker's current location

File: app/src/main/java/ch/epfl/balelecbud/model/PointOfInterestType.java [hunk 4]
```diff
@@ -1,16 +1,27 @@
 package ch.epfl.balelecbud.model;
 
+import androidx.annotation.NonNull;
+
+/**
+ * Enum modeling the different type of point of interest of the festival
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/model/PointOfInterestType.java:6]: Enum modeling the different types of points of interest of the festival

File: app/src/main/java/ch/epfl/balelecbud/utility/cloudMessaging/CloudMessagingService.java [hunk 5]
```diff
@@ -13,16 +13,22 @@
...
 import ch.epfl.balelecbud.BalelecbudApplication;
 import ch.epfl.balelecbud.R;
-import ch.epfl.balelecbud.utility.http.HttpPostRequest;
 import ch.epfl.balelecbud.utility.notifications.NotificationMessage;
 
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppContext;
 
-public class CloudMessagingService extends FirebaseMessagingService implements MessagingService {
+/**
+ * Implementation of {@code MessaginService} with Firebase Cloud Messaging
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/utility/cloudMessaging/CloudMessagingService.java:21]: {@code MessagingService}

File: app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyGeoClause.java [hunk 6]
```diff
@@ -1,13 +1,22 @@
 package ch.epfl.balelecbud.utility.database.query;
 
-public class MyGeoClause implements MyClause {
+/**
+ * Class modeling a geographic claus represented by a disc
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyGeoClause.java:4]: clause* represented by a circular area

File: app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyGeoClause.java [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
     private final double searchRadius;
 
-
+    /**
+     * Constructor for geographic claus
+     *
+     * @param centreLatitude  the latitude of the center
+     * @param centreLongitude the longitude of the center
+     * @param searchRadius    the radius of the disc
+     */
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyGeoClause.java:19]: clause* / radius of the area*

File: app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyQuery.java [hunk 8]
```diff
@@ -7,7 +7,10 @@
 
 import ch.epfl.balelecbud.utility.database.Database;
 
-public class  MyQuery {
+/**
+ * Class modeling normal claus
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/utility/database/query/MyQuery.java:11]: Santa Clause

File: app/src/main/java/ch/epfl/balelecbud/utility/location/LocationService.java [hunk 9]
```diff
@@ -15,7 +15,10 @@
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppAuthenticator;
 import static ch.epfl.balelecbud.BalelecbudApplication.getAppDatabase;
 
-public class LocationService extends IntentService {
+/**
+ * A service process location updates
```
Comment (karimassi) [inline app/src/main/java/ch/epfl/balelecbud/utility/location/LocationService.java:19]: A service that processes location updated

File: app/src/main/java/ch/epfl/balelecbud/utility/recyclerViews/RefreshableRecyclerViewAdapter.java [hunk 10]
```diff
@@ -13,16 +12,19 @@
...
 import ch.epfl.balelecbud.BalelecbudApplication;
 import ch.epfl.balelecbud.R;
 import ch.epfl.balelecbud.utility.DateFormatter;
 import ch.epfl.balelecbud.utility.database.Database;
 
-public class RefreshableRecyclerViewAdapter<A, B extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<B> {
+/**
+ * Generic adapter for refreshable recycler view
+ * @param <A> the type of the data displayed in the recycler view
+ * @param <B> the type of the
```
Comment (axelmarmet) [inline app/src/main/java/ch/epfl/balelecbud/utility/recyclerViews/RefreshableRecyclerViewAdapter.java:25]:
```suggestion
 * @param <B> the type of the view holder for the displayed data
```

File: app/src/main/java/ch/epfl/balelecbud/view/emergency/EmergencyInformationData.java [hunk 11]
```diff
@@ -11,12 +11,15 @@
 import ch.epfl.balelecbud.utility.recyclerViews.OnRecyclerViewInteractionListener;
 import ch.epfl.balelecbud.utility.recyclerViews.RecyclerViewData;
 
-public class EmergencyInformationData extends RecyclerViewData<EmergencyInformation, EmergencyInformationHolder> {
+/**
+ *
```
Comment (axelmarmet) [inline app/src/main/java/ch/epfl/balelecbud/view/emergency/EmergencyInformationData.java:15]:
```suggestion
 * Implementation of RecyclerViewData to display emergency informations
```

**Pull request merged**